canals/pipeline/pipeline.py

```diff
@@ -278,7 +277,7 @@ def warm_up(self):
                 logger.info("Warming up component %s...", node)
                 self.graph.nodes[node]["instance"].warm_up()
 
-    def run(self, data: Dict[str, ComponentInput], debug: bool = False) -> Dict[str, Any]:
+    def run(self, data: Dict[str, type], debug: bool = False) -> Dict[str, Any]:
```

Comment:
I'm not sure about using `Any` to be fair. It can't be anything, it must be a dataclass.

The problem is a `dataclass` is simply a `class` with `__dataclass_fields__` set to a non-empty list, so there's really no way of defining its type.

But I do agree that it must identical to the return type.

Reply:
Thinking about it probably marking it as `Any` is the best choice.

Reply:
yeah I also wish one could type dataclass... Let's use Any for now, I still hope I can add a thin layer of typing on top of this in another PR soon